pageserver/Cargo.toml

```diff
@@ -10,10 +10,11 @@ edition = "2018"
 chrono = "0.4.19"
 rand = "0.8.3"
 regex = "1.4.5"
-bytes = "1.0.1"
+bytes = { version = "1.0.1", features = ['serde'] }
```

Comment:
> I'm curious what this does.

With this, `Bytes` derives Serializable and Deserializable, so that `Bytes` can be used in structs that are marked with `#[derive(Serialize, Deserialize)]`.

> I'm a little wary of expanding our use of `Bytes`/`BytesMut` because they don't mate well with parts of the standard library.

I find them very handy. A page image can be read from rocksdb/sled, or it can be reconstructed from WAL. Reference counted Bytes makes it easy to pass it around the system, without worrying about ownership. I think the alternative would be `Arc<Vec<u8>>`, but might as well use `Bytes`.

BTW, 'sled' has a similar type called [IVec](https://docs.rs/sled/0.34.6/sled/struct.IVec.html).

Reply:
That seems kind of sketchy to me. Why would I want a reference-counted type inside a serializable struct?

Our code over-uses `Bytes` and `BytesMut`, and I think that this is a crutch that allows people to skip over understanding basic Rust arrays, `Vec<u8>`, `Box<[u8]>` etc. and those are really important concepts.

> easy to pass it around the system, without worrying about ownership

When people are learning the language they should spend some time worrying about ownership. They should care whether a reference is shared or mutable, whether a type is `Copy` or refcounted, etc.